template/tools/discord/search_tool.py

```diff
@@ -0,0 +1,87 @@
+import json
+from typing import Type
+from pydantic import BaseModel, Field
+from starlette.types import Send
+from template.services.discord_prompt_analyzer import DiscordPromptAnalyzer
+from template.tools.base import BaseTool
+import bittensor as bt
+
+
+class SearchDiscordToolSchema(BaseModel):
+    query: str = Field(
+        ...,
+        description="The search query for Discord messages.",
+    )
+
+
+class SearchDiscordTool(BaseTool):
+    """Tool that searches for messages in Discord based on a query."""
+
+    name = "Search Discord"
+    slug = "search_discord"
```

Comment:
rename to DiscordSearchTool to keep same naming structure for tool

and name also "Discord Search"